src/parsec.rs

```diff
@@ -90,6 +131,11 @@ impl ParsecMap {
         pub_id: id::PublicId,
         log_ident: &LogIdent,
     ) -> bool {
+        // Increase the size before fetching the parsec to satisfy the borrow checker
+        if self.last_version() == msg_version && self.map.contains_key(&msg_version) {
```

Comment:
the comment and the if clause can be put into the `count_size` function?
there is duplication of the code.
I think you only need to pass the `msg_version` into `count_size` function ?

Reply:
Pushed a fixup that moves the guard inside count_size. Although I think strictly speaking it's not counting the size anymore...now